ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -235,6 +238,23 @@ TransientQueryMetadata buildTransientQuery(
     return Optional.empty();
   }
 
+  private boolean isQuerySupported(final KeyFormat pullQueryKeyFormat,
+                                   final LogicalSchema schema) {
+    final boolean isFormat = (pullQueryKeyFormat.getFormatInfo().getFormat()
+            .equalsIgnoreCase("KAFKA")
+            || pullQueryKeyFormat.getFormatInfo().getFormat().equalsIgnoreCase("DELIMITED"));
+    boolean notSupported = (isFormat && ((schema.key().size() != 1)
+            || schema.key().get(0).type().baseType() == SqlBaseType.STRUCT
```

Comment:
Do you need to check the other keys beyond the 0th entry?

Reply:
Only need to check the 0th entry.